packages/ckeditor5-ui/src/toolbar/block/blocktoolbar.ts

```diff
@@ -630,35 +621,46 @@ export default class BlockToolbar extends Plugin {
 /**
  * Returns the closest scrollable parent element of the given element.
  *
- * @param element - The element for which to find the closest scrollable parent.
+ * @param element The element for which to find the closest scrollable parent.
  * @returns The closest scrollable parent element, or `null` if no scrollable parent is found.
  */
-function closestScrollableParent( element: HTMLElement ): HTMLElement | null {
-	if ( isScrollable( element ) ) {
-		return element;
-	}
+function closestScrollableElement( element: HTMLElement ): HTMLElement | null {
+	let currentElement: HTMLElement | null = element;
 
-	let parent = element.parentElement;
-	let depth = 0;
+	do {
```

Comment:
I wonder if some of this logic could be replaced with methods of the `Rect`, for example this one: https://github.com/ckeditor/ckeditor5/blob/08270aafcc6e7e4fb4f4a4d2ebbc7451f031d303/packages/ckeditor5-utils/src/dom/rect.ts#L254.

Reply:
I'll check that.

Reply:
Hm, it looks like it works ootb!